File: SDKs/linode.com/4.97.2/python/src/sdk/models/operations/getpersonalaccesstoken.py
from dataclasses import dataclass, field
from typing import List,Optional
from dataclasses_json import dataclass_json
from sdk import utils
from sdk.models import shared


@dataclass
class GetPersonalAccessTokenPathParams:
    token_id: int = field(metadata={'path_param': { 'field_name': 'tokenId', 'style': 'simple', 'explode': False }})
    

@dataclass
class GetPersonalAccessTokenSecurity:
    oauth: Optional[shared.SchemeOauth] = field(default=None, metadata={'security': { 'scheme': True, 'type': 'oauth2' }})
    personal_access_token: Optional[shared.SchemePersonalAccessToken] = field(default=None, metadata={'security': { 'scheme': True, 'type': 'http', 'sub_type': 'bearer' }})
    

@dataclass_json
@dataclass
class GetPersonalAccessTokenDefaultApplicationJSON:
    errors: Optional[List[shared.ErrorObject]] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('errors') }})
    

@dataclass
class GetPersonalAccessTokenRequest:
    path_params: GetPersonalAccessTokenPathParams = field()
    security: GetPersonalAccessTokenSecurity = field()
    

@dataclass
class GetPersonalAccessTokenResponse:
    content_type: str = field()
    status_code: int = field()
    personal_access_token: Optional[shared.PersonalAccessToken] = field(default=None)
    get_personal_access_token_default_application_json_object: Optional[GetPersonalAccessTokenDefaultApplicationJSON] = field(default=None)
    

File: SDKs/linode.com/4.97.2/typescript/src/sdk/models/operations/getpersonalaccesstoken.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import * as shared from "../shared";



export class GetPersonalAccessTokenPathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=tokenId" })
  tokenId: number;
}


export class GetPersonalAccessTokenSecurity extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "security, scheme=true;type=http;subtype=bearer" })
  personalAccessToken?: shared.SchemePersonalAccessToken;

  @SpeakeasyMetadata({ data: "security, scheme=true;type=oauth2" })
  oauth?: shared.SchemeOauth;
}


export class GetPersonalAccessTokenDefaultApplicationJson extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=errors", elemType: shared.ErrorObject })
  errors?: shared.ErrorObject[];
}


export class GetPersonalAccessTokenRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  pathParams: GetPersonalAccessTokenPathParams;

  @SpeakeasyMetadata()
  security: GetPersonalAccessTokenSecurity;
}


export class GetPersonalAccessTokenResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  personalAccessToken?: shared.PersonalAccessToken;

  @SpeakeasyMetadata()
  statusCode: number;

  @SpeakeasyMetadata()
  getPersonalAccessTokenDefaultApplicationJsonObject?: GetPersonalAccessTokenDefaultApplicationJson;
}
